VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Activities/Productizer/Operations/JobApplicantProfile/UpdateJobApplicantProfile.cs

```diff
@@ -48,19 +51,78 @@ public void SetAuth(Guid? userDatabaseId)
         {
             UserId = userDatabaseId;
         }
+
+        private sealed class WorkPreferencesValidator : AbstractValidator<Request.WorkPreferenceValues>
+        {
+            public WorkPreferencesValidator()
+            {
+                RuleForEach(wp => wp.PreferredMunicipality)
+                    .Must(x => EnumUtilities.TryParseWithMemberName<Municipality>(x, out _));
+
+                RuleForEach(wp => wp.PreferredRegion)
+                    .Must(x => EnumUtilities.TryParseWithMemberName<Region>(x, out _));
+
+                RuleFor(x => x.TypeOfEmployment)
+                    .Must(x => EnumUtilities.TryParseWithMemberName<EmploymentType>(x!, out _))
+                    .When(x => !string.IsNullOrEmpty(x.TypeOfEmployment));
+
+                RuleFor(x => x.WorkingTime)
+                    .Must(x => EnumUtilities.TryParseWithMemberName<WorkingTime>(x!, out _))
+                    .When(x => !string.IsNullOrEmpty(x.WorkingTime));
+
+                RuleForEach(wp => wp.WorkingLanguage)
+                    .Must(x => EnumUtilities.TryParseWithMemberName<WorkingLanguage>(x, out _));
+            }
+        }
+
+        public sealed class OccupationsValidator : AbstractValidator<List<Request.Occupation>>
+        {
+            private readonly IOccupationsFlatRepository _occupationsFlatRepository;
+
+            public OccupationsValidator(IOccupationsFlatRepository occupationsFlatRepository)
+            {
+                _occupationsFlatRepository = occupationsFlatRepository;
+
+                RuleFor(occupations => occupations)
+                    .MustAsync(async (occupations, cancellationToken) =>
+                    {
+                        var knownOccupations = await _occupationsFlatRepository.GetAllOccupationsFlat();
```

Comment:
Lataakos nämä jostain muistista, ettei network requa tee joka kerta. (varmaan ite ollut tämä päällä joskus, ei muista 😅 )

Reply:
> e: no vähemmän laiskana kävin kurkkaa, niin kai se sinne jonnekin muistiin menee (tuolla oli joku TODO in-memory cachee liittyen.. mutta toiminnee)

Joops tuo tieto tulee codesets-riippuvuudesta ja se repository säilöö asian lambda-instanssin elinajan verran, eli muistista muistelee. Ne TODO-merkinnät voisi ehkä käsitellä kanssa jossain kohti näitä siivoiluja..